chapters/chap03-sec02.tex

```diff
@@ -18,17 +18,97 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
 \end{align*}
 
-From now on, we only consider \mimc{} defined over prime fields. In particular, the field $\FFx{\rCURVE}$ with elements written over $\secpar$ bits, over which \zksnark~operates.
-
 \subsubsection{Security parameters and analysis}\label{instantiation:mkhash:mimc-encryption:security}
 
-To ensure that the exponentiation leads to a permutation in $\FFx{\rCURVE}$, $\exponent{}$ of the form $\exponent{} = 2^t-1$ such that $\gcd(\exponent{}, \rCURVE-1) = 1$. To achieve a security of $\secpar$, we require $\rounds = \left\lceil \frac{\log_2 \rCURVE}{\log_2 \exponent{}} \right\rceil. $\footnote{We do not consider exponents of type $2^t+1$ since the polynomial representing $\mimcEnc$ would be sparse and as such, the number of rounds becomes constant (see~\cite[Section 5.3]{albrecht2016mimc}).}
+In the original design proposed in the $\mimc{}$ paper~\cite{albrecht2016mimc}, the round function is represented as a ``shifted'' permutation via a cubic map (i.e.~the round input message is added to the key and round constant - the ``shift'' -, and a map $x^{\exponent{}}$ permutes the element in the underlying field $\FFx{2^n}$, $n \in \NN$, where $\gcd(\exponent{}, 2^n - 1) = 1$). This function (which is a permutation and therefore invertible) acts as a substitution box (S-box) and brings non-linearity to the scheme, as usually required for security.
+
+In other sections of the paper, however, the $\mimc{}$ authors proposed generalizations to the initial design. These allow $\mimc{}$ to be used:
+\begin{itemize}
+    \item over prime fields of odd characteristic (i.e.~$\FFx{p}$, $p$ odd prime),
+    \item with different permutation polynomials (i.e.~using different exponents in the round function)
+\end{itemize}
+
+Understanding the relationships between these various settings is required in order to use $\mimc{}$ to operate over prime fields of odd characteristic with a non-cubic permutation monomial.
+
+Overall, for $\mimc{}$ to be considered secure, it is important that no attack in the literature (that may provide a significant speedup compared to ``exhaustive key search'') can succesfully be mounted by a $\ppt$ adversary. Two main families of attacks exist: statistical attacks and algebraic attacks.
+
+\paragraph{Statistical attacks.}
+
+In the ``Security Analysis'' section~\cite[Section 4.2]{albrecht2016mimc}, the authors explain that since the cubic function is an APN, linear attacks pose no threat to $\mimc{}$.
+
+We observe that this claim aligns with~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99}. In fact if $\mimc{}$ is operated over $\FFx{2^n}$, it is easy to see that the degree of the cubic map can be expressed as $3 = 2^t + 1$, where $t = 1$, and where $n$ and $1$ are trivially coprime. This case is covered by~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99} which confirms that the cubic function $S(x) = x^3$ over $\FFx{2^n}$ is an APN/2-uniform mapping, as desired for differential and linear cryptanalysis resistance.
```

Comment:
"APN/2-uniform mapping" -- could we simple explain what is that? If not, could we / should we assume that the reader is well familiar with symmetric cryptography?

Reply:
See: https://github.com/clearmatics/zeth-specifications/pull/6#discussion_r607895806

I'm happy to unroll the APN acronym, however, since this section is justifying some of the security claims around MiMC and its instantiation in the specs, I assume some domain knowledge here. If the reader is simply interested in implementing the protocol without bothering with the security proofs, they can still skip this section.